### PR TITLE
retry cluster policy updates in test logic

### PR DESCRIPTION
**Pull request by bparees**
workaround for https://github.com/openshift/origin/issues/17596

### CODE REVIEW

**Comment by bparees** [comment]
@enj @deads2k fyi.
/assign @jim-minter

File: test/extended/builds/forcepull.go [hunk 1]
```diff
@@ -43,10 +43,25 @@ func scrapeLogs(bldPrefix string, oc *exutil.CLI) {
 func checkPodFlag(bldPrefix string, oc *exutil.CLI) {
 	if bldPrefix == buildPrefixTC {
 		// grant access to the custom build strategy
-		err := oc.AsAdmin().Run("adm").Args("policy", "add-cluster-role-to-user", "system:build-strategy-custom", oc.Username()).Execute()
+		var err error
+		for i := 0; i < 5; i++ {
+			err = oc.AsAdmin().Run("adm").Args("policy", "add-cluster-role-to-user", "system:build-strategy-custom", oc.Username()).Execute()
```
Comment (enj) [inline test/extended/builds/forcepull.go]: Just do `oc create clusterrolebinding <name of test + curent time> --clusterrole=<...> --user=<...>` to skip the conflict.

File: test/extended/builds/forcepull.go [hunk 2]
```diff
@@ -43,10 +43,25 @@ func scrapeLogs(bldPrefix string, oc *exutil.CLI) {
...
+			}
+			break
+		}
 		o.Expect(err).NotTo(o.HaveOccurred())
+
 		defer func() {
-			err = oc.AsAdmin().Run("adm").Args("policy", "remove-cluster-role-from-user", "system:build-strategy-custom", oc.Username()).Execute()
+			var err error
+			for i := 0; i < 5; i++ {
+				err = oc.AsAdmin().Run("adm").Args("policy", "remove-cluster-role-from-user", "system:build-strategy-custom", oc.Username()).Execute()
```
Comment (enj) [inline test/extended/builds/forcepull.go]: Delete the cluster role binding you created above.

**Comment by enj** [comment]
/hold

Lets not brute force our way through this.

**Comment by bparees** [comment]
@enj if you're going to hold my PR I need you to respond my questions.

**Comment by enj** [comment]
> creating unique timestamped roles isn't "brute forcing our way through this"?

Sounds a lot better than manually retrying in a for loop.

> the oc adm command is fundamentally flawed and should be fixed (hence why this PR isn't closing the open issue, and I would expect to be able to revert this workaround PR in the future).

Sure, but @deads2k seem against changes to those commands, and I have not had the time.

> "Go create a unique rolebinding every time you want to grant a user access to something" isn't what we'd tell our customers, why would we do it in a test?

This is what upstream does, and what we are moving towards.

**Comment by deads2k** [comment]
> the oc adm command is fundamentally flawed and should be fixed (hence why this PR isn't closing the open issue, and I would expect to be able to revert this workaround PR in the future).

The solution is probably for `oc adm policy add/remove` to be deprecated.  We're going to follow upstream by creating unique clusterrolebindings and removing unique ones as well.  `oc adm policy` is inconsistent with upstream RBAC, apply, and reconcile patterns.

**Comment by deads2k** [comment]
> "Go create a unique rolebinding every time you want to grant a user access to something" isn't what we'd tell our customers, why would we do it in a test?

Yes, it is.

/hold

**Comment by bparees** [comment]
fix the oc adm command, then, guys.

**Comment by deads2k** [comment]
> fix the oc adm command, then, guys.

I can mark it deprecated pretty quickly, but doesn't that leave you where you are?

**Comment by bparees** [comment]
> I can mark it deprecated pretty quickly, but doesn't that leave you where you are?

If you get it marked deprecated i'll see a reason to move test code away from using it and onto the pattern we are forcing on our customers.  Of course you'll also need to update all the doc+examples when you deprecate it.  Ansible installer uses it too, so don't forget that.

I also still think telling customers to create a binding per user is unfriendly and will be the next etcd object explosion problem we'll be fighting.

**Pull request closed without merging**